### PR TITLE
Rework #1470 part 2

### LINKED ISSUE #1470: Change Score.compute to take defaultBranch version and release count
It currently takes all versions but only uses the release count and some default version info. We can make the call into `Score.compute` in `updatePackages` (`Common.swift`) cheaper by not loading all versions.

### PR DESCRIPTION
**Pull request by finestructure**
Merge after #2135 

Fixes #1470

- Analyze.updateLatestVersion → a/a, returning the updated versions
- `analyze` is calling a new function `updateScore` using the previously loaded `Version`s
- `Common.updatePackages` is not calling `pkg.$versions.load(on: db)` anymore, fixing #1470
- Test updates according to above changes

### CODE REVIEW

**Review by daveverwer — APPROVED** [review]

**Comment by finestructure** [comment]
It's flatlining again on this revision:

CleanShot 2022-11-18 at 12 22 34@2x

**Comment by finestructure** [comment]
Deploying ea46568e7024f02de02b526c3e69596dfd9c0c2f to see if that tweak in `updateLatestVersions` makes a difference.

**Comment by finestructure** [comment]
Stuck again:

CleanShot 2022-11-21 at 11 35 35@2x

The good news is this seems to be failing quickly and only for analysis. So even if this isn't fixable, we'd still have a win by avoiding the version fetch in ingestion.

**Comment by finestructure** [comment]
I have a theory why this might be failing in analysis and I'm preparing to test this. We're processing batches of 25 packages and in the live analysis we only send the packages to the update function which refetches the versions for each version. In this updated version we send 25 packages _and all their versions_ on each invocation, avoiding the re-fetch. I haven't actually checked when the process failed but I could imagine that for certain batches we're hitting a RAM limit with this change.

I'll be testing this by reducing the batch size (and increasing the frequency accordingly) to see if this fixes the problem.

**Comment by finestructure** [comment]
Running at batch size 5, sleep 4 is also stalling. Don't think it's that.

**Comment by finestructure** [comment]
Ok, I think I've finally fixed this by moving the score update to analysis entirely. We were doing it as part of `updatePackages` which was called from both ingestion and analysis and yesterday I realised that there's no point actually trying to do it there if we're then adding an overload that updates the score in one case (when called from analysis) and not in the other (when called from ingestion).

I'm not sure _why_ the process stalled when we passed along the versions but this is the better solution anyway. I deployed 6289dcc1a9692b2ce1b28675c918f7a8ed122e81 from the branch this morning and it's been working fine so far:

CleanShot 2022-11-22 at 08 03 25@2x

There's a dramatic improvement in ingestion processing time to a bit less than half.

**Comment by finestructure** [comment]
Ready for review but let's not merge it straight away and give the deployed version more time.

**Comment by finestructure** [comment]
I want to make sure the mon-update/rollback works ok on dev before tagging that version and deploying it. Then I'll merge this when things have settled down.

**Comment by finestructure** [comment]
We have another "lockup" this morning on dev:

CleanShot 2022-11-25 at 08 19 54@2x

However, dev is running b2f22b592e5511dc1223e36ed291851c471d7a82, which is current `main`, i.e. does not include the changes here. So this issue isn't actually related to these changes.

The only difference between b2f22b592e5511dc1223e36ed291851c471d7a82 and 2.79.10, which is running on prod without any issues, is minimal:

[2.79.10..b2f22b592e5511dc1223e36ed291851c471d7a82](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/compare/2.79.10..b2f22b592e5511dc1223e36ed291851c471d7a82)

**Pull request merged**
